firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt

```diff
@@ -79,7 +86,27 @@ internal constructor(
       apiKey,
       modelName,
       requestOptions.toInternal(),
-      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}"
+      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}",
+      object : HeaderProvider {
+        override val timeout: Duration
+          get() = 10.seconds
+
+        override suspend fun generateHeaders(): Map<String, String> {
+          if (appCheckTokenProvider == null) {
+            Log.w(TAG, "AppCheck not registered, skipping")
+            return emptyMap()
+          }
+          val mutableHeaderMap: MutableMap<String, String> = mutableMapOf()
```

Comment:
Either of these is completely fine, just wanted to make sure you know that you can also do this:

```suggestion
          val mutableHeaderMap = mutableMapOf<String, String>()
```